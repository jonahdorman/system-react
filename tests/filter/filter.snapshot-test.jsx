import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Default from '../../examples/filter/default';
import NewFilter from '../../examples/filter/new';
import LockedFilter from '../../examples/filter/locked';
import PermanantFilter from '../../examples/filter/permanant';
import ErrorFilter from '../../examples/filter/error';

test('Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<Default />
	));
	expect(domTree).toMatchSnapshot();
});

test('NewFilter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<NewFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('LockedFilter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<LockedFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('Permanant Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<PermanantFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('Error Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<ErrorFilter />
	));
	expect(domTree).toMatchSnapshot();
});
