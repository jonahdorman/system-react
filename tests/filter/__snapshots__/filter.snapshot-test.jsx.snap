exports[`test Error Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveTextEditFilter="Edit filter:"
  assistiveTextEditFilterHeading="Choose filter criteria"
  id="sample-panel-filtering-show-me"
  isError={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveTextEditFilter="Edit filter:"
  assistiveTextEditFilterHeading="Choose filter criteria"
  id="sample-panel-filtering-show-me"
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test LockedFilter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveTextEditFilter="Edit filter:"
  assistiveTextEditFilterHeading="Choose filter criteria"
  id="sample-panel-filtering-show-me"
  isLocked={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test NewFilter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveTextEditFilter="Edit filter:"
  assistiveTextEditFilterHeading="Choose filter criteria"
  id="sample-panel-filtering-show-me"
  isNew={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test Permanant Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveTextEditFilter="Edit filter:"
  assistiveTextEditFilterHeading="Choose filter criteria"
  id="sample-panel-filtering-show-me"
  isPermanent={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;
